summingbird-online/src/main/scala/com/twitter/summingbird/online/executor/AsyncBase.scala

```diff
@@ -135,10 +136,12 @@ abstract class AsyncBase[I, O, S, D, RC](maxWaitingFutures: MaxWaitingFutures, m
       val toClear = outstandingFutures.size - maxWaitingFuturesCount
       if (toClear > 0) {
         try {
-          Await.ready(AsyncBase.waitN(outstandingFutures, toClear), maxWaitingTime.get)
+          val dequeuedFutures = outstandingFutures.toSeq
+          Await.ready(AsyncBase.waitN(dequeuedFutures, toClear), maxWaitingTime.get)
+          outstandingFutures.putAll(dequeuedFutures.filter(_.isDefined))
```

Comment:
you want `outstandingFutures.putAll(dequeuedFutures.filterNot(_.isDefined))` right? You don't want to requeue if the future is complete, do you?